core/web-assets/src/main/assets/js/apps/onms-requisitions/lib/scripts/model/RequisitionNode.js

```diff
@@ -156,8 +157,43 @@ const RequisitionNode = function RequisitionNode(foreignSource, node, isDeployed
    */
   self.assets = [];
 
+  /**
+   * @description The array of requisition meta-data entries
+   * @ngdoc property
+   * @name RequisitionNode#requisitionMetaData
+   * @propertyOf RequisitionNode
+   * @returns {object} The requisition meta-data entries
+   */
+  self.requisitionMetaData = [];
+
+  /**
+   * @description The array of other meta-data entries
+   * @ngdoc property
+   * @name RequisitionNode#otherMetaData
+   * @propertyOf RequisitionNode
+   * @returns {object} The other meta-data entries
+   */
+  self.otherMetaData = {};
+
   angular.forEach(node['interface'], function(intf) {
-    self.interfaces.push(new RequisitionInterface(intf));
+      self.interfaces.push(new RequisitionInterface(intf));
+  });
+
+  angular.forEach(node['meta-data'], function(entry) {
+    if (entry.context === 'requisition') {
+      self.requisitionMetaData.push({
+        'key': entry.key,
+        'value': entry.value,
+      });
+    } else {
+      if (!_.has(self.otherMetaData, entry.context)) {
+        self.otherMetaData[entry.context] = []
+      }
+      self.otherMetaData[entry.context].push({
+        'key': entry.key,
+        'value': entry.value,
+      });
+    }
```

Comment:
This whole block is duplicated a few times, can we somhow reduce that?

Reply:
Not that I'm aware off. All things I've tried to do so are much more complex than these few lines of code.

Reply:
`angular.forEach(...)` could be rewritten, by something like 
```
 if (isArray(x)) {
   x.forEach(function(entry) {... });
};
```
or just `x.forEach(...)`. 

Besides this, putting it into its own file, maybe `Utils` shouldn't be that hard or do I miss something?

Reply:
I've moved the duplicated code to a separate model class `RequisitionMetaData.js` whis also de-duplicates the code generating the DTO Model from the meta-data.